scripts/update_from_application_services.py

```diff
@@ -1,185 +1,243 @@
+PACKAGE_SWIFT   = ROOT_DIR / "MozillaRustComponents" / "Package.swift"
+# Latest nightly.json produced by application‑services CI
+NIGHTLY_JSON_URL = (
+    "https://firefox-ci-tc.services.mozilla.com/api/index/v1/"
+    "task/project.application-services.v2.nightly.latest/artifacts/"
+    "public%2Fbuild%2Fnightly.json"
+)
+
+
+def main() -> None:
+    args     = parse_args()
+    version  = VersionInfo(args.version)
+    BRANCH   = "rcs-auto-update"
+    TITLE    = f"Nightly auto-update ({version.swift_version})"
+
+    # Ensure we have the latest remote copy of the update branch (if any)
+    subprocess.run(["git", "fetch", args.remote, BRANCH], check=False)
+
+    # Create (or reset) our working branch
+    remote_ref = f"refs/remotes/{args.remote}/{BRANCH}"
+    if subprocess.run(
+        ["git", "rev-parse", "--verify", remote_ref],
+        stdout=subprocess.DEVNULL,
+    ).returncode == 0:
+        subprocess.check_call(["git", "checkout", "-B", BRANCH, remote_ref])
+    else:
+        subprocess.check_call(["git", "checkout", "-B", BRANCH, args.base])
 
+    # Apply the nightly update
     update_source(version)
+
     if not repo_has_changes():
         print("No changes detected, quitting")
-        sys.exit(0)
-    # Make a branch for the PR
-    subprocess.check_call(["git", "checkout", "-B", BRANCH])
-    subprocess.check_call([
-        "git",
-        "commit",
-        "--author",
-        "Firefox Sync Engineering <sync-team@mozilla.com>",
-        "--message",
-        TITLE
-    ])
+        return
+
+    # Stage everything
+    subprocess.check_call(["git", "add", "-A", "MozillaRustComponents"])
+
+    # Actually check if we have anything locally
+    has_staged = (
+        subprocess.run(
+            ["git", "diff", "--cached", "--quiet"],
+            stdout=subprocess.DEVNULL,
+            stderr=subprocess.DEVNULL,
+        ).returncode
+        != 0
+    )
+
+    if not has_staged:
+        print("No changes detected, quitting")
+        return
+
+    # Commit
+    subprocess.check_call(
+        [
+            "git",
+            "commit",
+            "--author",
+            "Firefox Sync Engineering <sync-team@mozilla.com>",
+            "--message",
+            TITLE,
+        ]
+    )
+
+    # Push and open / update the PR
     if args.push:
-        subprocess.check_call(["git", "push", "--force-with-lease", "-u", args.remote, BRANCH])
+        subprocess.check_call(
+            ["git", "push", "--force-with-lease", "-u", args.remote, BRANCH]
+        )
 
         result = subprocess.run(
             ["gh", "pr", "view", BRANCH, "--json", "state", "--jq", ".state"],
             text=True,
             capture_output=True,
         )
-        # If the branch already has an open PR, nothing else to do
+
         if result.returncode == 0 and result.stdout.strip() == "OPEN":
-            print("PR already open – branch updated in place")
+            print("PR already open, branch updated in place")
```

Comment:
We will need to update the title and body to keep things in sync or maybe just skip adding the version numbers. Something like this should do the trick:

```python
subprocess.check_call(
        [
            "gh",
            "pr",
            "edit",
            BRANCH,
            "--title",
            TITLE,
            "--body",
            f"Automatically updated nightly build for `{version.swift_version}`.",
        ]
    )
```